dump/main.c

```diff
@@ -140,9 +140,68 @@ static int dump_full(const char* spec, bool used_sectors)
 	return 0;
 }
 
+static int dump_file(const char* spec, const char* path)
+{
+	struct exfat ef;
+	struct exfat_node* node;
+	cluster_t cluster, next_cluster;
+	cluster_t start_cluster = 0;
+	off_t remainder;
+	off_t extent_size;
+	off_t lsize;
+	int rc;
+
+	if (exfat_mount(&ef, spec, "ro") != 0)
+		return 1;
+
+	rc = exfat_lookup(&ef, &node, path);
+
+	cluster = exfat_advance_cluster(&ef, node, 0);
+	if (CLUSTER_INVALID(cluster))
+	{
+		exfat_unmount(&ef);
+		exfat_error("invalid cluster 0x%x while reading", cluster);
+		return -1;
+	}
+
+	remainder = node->size;
+	while (remainder > 0)
+	{
+		if (start_cluster == 0)
+		{
+			/* Start a new extent */
+			start_cluster = cluster;
+			extent_size = 0;
+		}
+
+		lsize = MIN(CLUSTER_SIZE(*ef.sb), remainder);
+		extent_size += lsize;
+
+		if (remainder <= CLUSTER_SIZE(*ef.sb))
+		{
+			/* We're at the end of the file, print last extent */
+			printf("%ld %ld\n", exfat_c2o(&ef, start_cluster), extent_size);
```

Comment:
Printed values will be truncated on 32-bit machines. Use `PRIu64` to print `off_t` types.